docs/install/installing-istio.md

```diff
@@ -0,0 +1,281 @@
+
+## Before you begin
+
+You need:
+- A Kubernetes cluster created
+- [`helm`](https://helm.sh/) installed
+
+## Installing Istio
+
+When you install Istio, there are a couple of different steps, and a few options
+depending on your goals. For a basic Istio installation suitable for most use
+cases, see the [Default Istio installation](#default-istio-instllation)
+instructions. Those steps will get you up and running quickly without having to
+make decisions about Istio. To customize Istio your Istion installion for use
+with Knative, see the [Custom Istio installation](#customizing-your-installation) instructions.
+
+### Default Istio installation
+The following steps install a default version of Istio that is appropriate for
+most Knative use cases.
+
+1. Enter the following commands to download Istio:
+   ```shell
+   # Download and unpack Istio
+   export ISTIO_VERSION=1.1.3
+   curl -L https://git.io/getLatestIstio | sh -
+   cd istio-${ISTIO_VERSION}
+   ```
+
+1. Enter the following command to install the Istio CRDs first:
+   ```shell
+   for i in install/kubernetes/helm/istio-init/files/crd*yaml; do kubectl apply -f $i; done
+   ```
+   Wait a few seconds for the CRDs to be committed in the Kubernetes API-server,
+   then continue with these instructions.
+
+1. Enter the following command to install Istio:
+   ```shell
+   # A template with sidecar injection enabled.
+   helm template --namespace=istio-system \
+    --set sidecarInjectorWebhook.enabled=true \
+    --set sidecarInjectorWebhook.enableNamespacesByDefault=true \
+    --set global.proxy.autoInject=disabled \
+    --set global.disablePolicyChecks=true \
+    --set prometheus.enabled=false \
+    `# Disable mixer prometheus adapter to remove istio default metrics.` \
+    --set mixer.adapters.prometheus.enabled=false \
+    `# Disable mixer policy check, since in our template we set no policy.` \
+    --set global.disablePolicyChecks=true \
+    `# Set gateway pods to 1 to sidestep eventual consistency / readiness problems.` \
+    --set gateways.istio-ingressgateway.autoscaleMin=1 \
+    --set gateways.istio-ingressgateway.autoscaleMax=1 \
+    --set gateways.istio-ingressgateway.resources.requests.cpu=500m \
+    --set gateways.istio-ingressgateway.resources.requests.memory=256Mi \
+    `# More pilot replicas for better scale` \
+    --set pilot.autoscaleMin=2 \
+    `# Set pilot trace sampling to 100%` \
+    --set pilot.traceSampling=100 \
+    install/kubernetes/helm/istio \
+    `# Removing trailing whitespaces to make automation happy` \
+    | sed 's/[ \t]*$//' \
+    > ./istio.yaml
+
+   kubectl apply -f istio.yaml
+   ```
+
+This default installation enables [automatic sidecar injection][1].
+
+## Customizing your installation
+
+You can easily customize your Istio installation with `helm`. The below sections
+cover a few useful customizations and their purpose.
+
+### Installing Istio with sidecar injection
+
+If you need Istio service mesh, and want to enable it by
```

Comment:
@ZhiminXiang 
can we add some info on why someone might need Istio service mesh? Would be helpful to give an example use case for doing so.

Reply:
Istio service mesh provides below benefits for users:
1. [mTLS](https://istio.io/docs/tasks/security/mutual-tls/): with service mesh enabled, users can turn on mTLS, which will make service-to-service traffic within cluster secure.
2. [Authorization policy](https://istio.io/docs/tasks/security/authz-http/): with service mesh enabled, users can leverage Istio authorization policy, which controls the access of each service within service mesh based on Istio service role.